src/model.coffee

```diff
@@ -169,6 +169,24 @@ class Model extends Backbone.Model
       )
       .promise(options.returnValues.jqXhr)
 
+  destroy: (options = {}) ->
+    cleanUpAssociatedReferences = =>
+      for collectionName, collection of @storageManager.collections
+        for reference, modelName of collection.modelKlass.associations
+          associationKey = collection.modelKlass.associationDetails(reference).key
+          if _.isArray(modelName) && _.find(modelName, (m) => inflection.singularize(m) == @className())
```

Comment:
For clarity, perhaps `modelName` should be pluralized or named in such a way to reflect it's an array?